File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0,j=0;
        int sl = s.size();
        int tl = t.size();
        while(j<tl){
            if(i==sl){
                return true;
            }
            if(t[j]==s[i]){
                i++;
                j++;
            }
            else{
                j++;
            }
        }
        if(i==sl){
                return true;
        }
        else{
            return false;
        }
    }
};

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int sPnt = 0;

        if(t.length() < s.length()){
            return false;
        }

        if(s.length() == 0){
            return true;
        }

        for(int i=0; i<t.length(); i++){
            if(t.charAt(i) == s.charAt(sPnt)){
                sPnt++;
            }

            if(sPnt == s.length()){
                return true;
            }
        }

        return false;
    }
}